src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -688,46 +688,50 @@ Error get(const std::string& id, bool includeContents, boost::shared_ptr<SourceD
       }
    }
    
-   if (propertiesPath.exists())
-   {
-      // read the contents of the file
-      std::string properties;
-      Error error = readStringFromFile(propertiesPath,
-                                       &properties,
-                                       options().sourceLineEnding());
-      if (error)
-         return error;
+   if (!propertiesPath.exists())
+      return core::fileNotFoundError(propertiesPath, ERROR_LOCATION);
    
-      // parse the json
-      json::Value value;
-      if (value.parse(properties))
-      {
-         return systemError(boost::system::errc::invalid_argument,
-                            ERROR_LOCATION);
-      }
-      
-      // initialize doc from json
-      json::Object jsonDoc = value.getObject();
-      
-      // migration: if we have a 'contents' field, but no '-contents' side-car
-      // file, perform a one-time generation of that sidecar file from contents
-      error = attemptContentsMigration(jsonDoc, propertiesPath);
-      if (error)
-         LOG_ERROR(error);
-      
-      if (includeContents && !contents.empty())
-         jsonDoc["contents"] = contents;
-      
-      if (jsonDoc.find("contents") == jsonDoc.end())
-         jsonDoc["contents"] = std::string();
-      
-      return pDoc->readFromJson(&jsonDoc);
+   // read the contents of the file
+   std::string properties;
+   Error error = readStringFromFile(propertiesPath,
+                                    &properties,
+                                    options().sourceLineEnding());
+   if (error)
+      return error;
+   
+   // parse the json
+   json::Value value;
+   if (value.parse(properties))
+   {
+      Error error = systemError(boost::system::errc::invalid_argument, ERROR_LOCATION);
+      error.addProperty("path", propertiesPath);
+      return error;
    }
-   else
+   
+   // validate we got a JSON object
+   if (!value.isObject())
```

Comment:
This check is necessary as otherwise files in the source database containing text that is valid JSON but _not_ a JSON object could lead to a crash. (While this in theory shouldn't ever happen, better safe than sorry)